sensor/common/scannerclient/grpc_client.go

```diff
@@ -212,45 +207,28 @@ func convertIndexReportToAnalysis(ir *v4.IndexReport) *ImageAnalysis {
 }
 
 func (c *v4Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg *types.Config) (*ImageAnalysis, error) {
-	imgSHA := utils.GetSHA(image)
-	if imgSHA == "" {
-		// TODO: ROX-19576: Is the assumption that image always have SHA correct?
-		return nil, fmt.Errorf("internal: V4 scanning of an image without SHA: %s",
-			image.GetName().GetFullName())
-	}
-	hashID := fmt.Sprintf("/v4/containerimage/%s", imgSHA)
-	opts := []grpc.CallOption{grpc.WaitForReady(true)}
-	ir, err := c.GetIndexReport(ctx, &v4.GetIndexReportRequest{HashId: hashID}, opts...)
-	if err == nil {
-		return convertIndexReportToAnalysis(ir), nil
-	}
-	// Check if index was "not found", so we create.
-	if s, _ := status.FromError(err); s.Code() != codes.NotFound {
-		return nil, fmt.Errorf("get index report (hashid: %s): %w", hashID, err)
-	}
-	log.Infof("creating index report: %s", hashID)
-	scheme := "https"
+	var opts []name.Option
 	if cfg.Insecure {
-		scheme = "http"
+		opts = append(opts, name.Insecure)
 	}
-	req := &v4.CreateIndexReportRequest{
-		HashId: hashID,
-		ResourceLocator: &v4.CreateIndexReportRequest_ContainerImage{
-			ContainerImage: &v4.ContainerImageLocator{
-				Url:      fmt.Sprintf("%s://%s", scheme, image.GetName().GetFullName()),
-				Username: cfg.Username,
-				Password: cfg.Password,
-			},
-		},
+	n := fmt.Sprintf("%s/%s@%s", image.GetName().GetRegistry(), image.GetName().GetRemote(), utils.GetSHA(image))
+	ref, err := name.NewDigest(n, opts...)
+	if err != nil {
+		// TODO: ROX-19576: Is the assumption that images always have SHA correct?
+		return nil, fmt.Errorf("creating digest reference: %w", err)
 	}
-	ir, err = c.CreateIndexReport(ctx, req, opts...)
+	auth := authn.FromConfig(authn.AuthConfig{
+		Username: cfg.Username,
+		Password: cfg.Password,
+	})
```

Comment:
```suggestion
	auth := &authn.Basic{
		Username: cfg.Username,
		Password: cfg.Password,
	}
```

Reply:
e1f575e0

Reply:
I adopted the suggested code, so I assume this is resolved @RTann